src/js/messaging/containers/Folder.jsx

```diff
@@ -80,6 +143,16 @@ export class Folder extends React.Component {
     });
   }
 
+  handleSearch(searchParams) {
+    this.props.resetPagination();
```

Comment:
It's kind of convoluted how pagination works in this way. At this point, it simply changes the display of what page you're currently on and what pages the pagination controls lead to. Then when we send the query with the search filters, it uses whatever pagination we just reset to, but it's kind of dependent on the timing of that change in relation to the push below. Rather than using an action creator and modifying the store to achieve that purpose, it would be better to just use the default page we want (I'm guessing the first page) when we make the query. In fact, not specifying any page at all for the API request should get us the first page.

And now thinking about it, upon submitting a search, we probably want to reset the sort as well, since submitting a search would intuitively clear the current filters (in terms of what page you're on and what you're sorted by).

So, ultimately, we wouldn't want to use `this.props.page` or `this.props.sort` when building the query for the request and can altogether drop `page` and `sort` from the URL being pushed to history.